src/chains/ethereum/transaction/src/transaction-factory.ts

```diff
@@ -168,59 +169,91 @@ export class TransactionFactory {
     extra?: GanacheRawExtraTx
   ) {
     const txType = this.typeOfRaw(txData);
-    return this._fromData(
-      <TypedDatabasePayload>(
-        (txType === LegacyTransaction ? txData : txData.slice(1)) // if the type is at the front, remove it
-      ),
-      txType,
-      common,
-      extra
-    );
+    switch (txType) {
+      case TransactionType.EIP1559AccessList:
+        return EIP1559FeeMarketTransaction.fromTxData(
+          txData.slice(1) as EIP1559FeeMarketDatabasePayload,
+          common,
+          extra
+        );
+      case TransactionType.Legacy:
+        return LegacyTransaction.fromTxData(
+          txData as LegacyDatabasePayload,
+          common,
+          extra
+        );
+      case TransactionType.EIP2930AccessList:
+        return EIP2930AccessListTransaction.fromTxData(
+          txData.slice(1) as EIP2930AccessListDatabasePayload,
+          common,
+          extra
+        );
+      default:
+        throw new CodedError(
+          `Transactions with supplied type ${txType} not supported`,
+          JsonRpcErrorCode.METHOD_NOT_FOUND
+        );
+    }
   }
   /**
    * Create a transaction from a `txData` object
    *
+   * When transaction types are activated (EIP 2718) the txData will be checked
+   * for a transaction envelope (first byte < 192) before determining the
+   * decoding strategy, otherwise it will be decoded as a Legacy Transaction. If
+   * the transaction contains a transaction envelop, but EIP 2718 is not active
+   * decoding will fail and an exception will be thrown.
```

Comment:
It seems worthwhile to add an actual error for the case  where a transaction envelope is sent but EIP 2718 is not activated. Otherwise it's a random buffer related error from `decode`, which would be a bit confusing.